amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/AmplifyTools.groovy

```diff
@@ -0,0 +1,219 @@
+                def configText = inputConfigFile.text
+                def jsonSlurper = new groovy.json.JsonSlurper()
+                def configJson = jsonSlurper.parseText(configText)
+                profile = configJson.profile
+                accessKeyId = configJson.accessKeyId
+                secretAccessKey = configJson.secretAccessKeyId
+                region = configJson.region
+                envName = configJson.envName
+                syncEnabled = configJson.syncEnabled
+            }
+        }
+
+        project.task('datastoreSync') {
+            def transformConfFile = project.file('amplify/backend/api/amplifyDatasource/transform.conf.json')
+            if (project.file('amplify/backend/api').exists()) {
+                new File('amplify/backend/api').eachFileRecurse(groovy.io.FileType.FILES) {
+                    if (it.name.endsWith('transform.conf.json')) {
+                        transformConfFile = project.file(it)
+                    }
+                }
+            }
+            if (transformConfFile.isFile()) {
+                def tranformConfText = transformConfFile.text
+                def jsonSlurper = new groovy.json.JsonSlurper()
+                def transformConfJson = jsonSlurper.parseText(tranformConfText)
+
+                def resolverConfigMap = ['ResolverConfig':
+                                                 ['project':
+                                                          ['ConflictHandler'  : 'AUTOMERGE',
+                                                           'ConflictDetection': 'VERSION']
+                                                 ]
+                ]
+                if (!syncEnabled) {
+                    transformConfJson.remove('ResolverConfig')
+                } else if (!transformConfJson.ResolverConfig) {
+                    transformConfJson << resolverConfigMap
+                }
+                def transformConfJsonStr = groovy.json.JsonOutput.toJson(transformConfJson)
+                def transformConfJsonStrPretty = groovy.json.JsonOutput.prettyPrint(transformConfJsonStr)
+                transformConfFile.write(transformConfJsonStrPretty)
+            }
+        }
+
+        project.task('modelgen') {
+            doLast {
+                project.exec {
+                    commandLine 'amplify', 'codegen', 'model'
+                }
+            }
+        }
+
+        project.task('amplifyPush') {
+            def AWSCLOUDFORMATIONCONFIG
+            if (!accessKeyId || !secretAccessKey || !region) {
+                AWSCLOUDFORMATIONCONFIG = """{\
+\"configLevel\":\"project\",\
+\"useProfile\":true,\
+\"profileName\":\"$profile\"\
+}"""
+            } else {
+                AWSCLOUDFORMATIONCONFIG = """{\
+\"configLevel\":\"project\",\
+\"useProfile\":true,\
+\"profileName\":\"$profile\",\
+\"accessKeyId\":\"$accessKeyId\",\
+\"secretAccessKey\":\"$secretAccessKey\",\
+\"region\":\"$region\"\
+}"""
+            }
```

Comment:
This is not a very maintainable way to build JSON. I don't know how to do this in Groovy, but in Java, you could something like:
```
new JSONObject()
    .put("key", new JSONObject()
        .put("innerKey", "value")
        .put("innerKey2", new JSONArray(Arrays.asList("val1", "val2")))
   );
```
Or, actually, here's a page dedicated to the topic: https://groovy-lang.org/json.html

Reply:
+1 `JsonSlurper` would be the groovy way of doing it.